### PR TITLE
Fixes #6325 Detect existing fetchpriority attribute when spaces around equal sign

### LINKED ISSUE #6325: Pre-existing fetchpriority won't be detected if there are spaces around the equal sign
**Before submitting an issue please check that you’ve completed the following steps:**
- Made sure you’re on the latest version :heavy_check_mark: 
- Used the search feature to ensure that the bug hasn’t been reported before :heavy_check_mark: 

**Describe the bug**

We won't be adding the `fetchpriority` property if one already exists:
https://github.com/wp-media/wp-rocket/blob/134ac6aa7bb5ce58b0fa6ac1b816f79acc21dc8b/inc/Engine/Media/AboveTheFold/Frontend/Controller.php#L135

However, if it's added with spaces before of after the `=` sign like so:
```html
<img src="https://example.com/dummy.png" fetchpriority = "high">
```

we will not detect it and add our own on top of it.

See the current RegEx in action: https://regex101.com/r/cqHgtE/1

**To Reproduce**
1. Add an LCP image to a page using the example code provided earlier.
2. After LCP has been processed by the SaaS, check the source code of the page.
3. Duplicate entries of the `fetchpriority` property will be there.

**Expected behavior**

The `fetchpriority` property should be correctly detected even if there are spaces before/after the `=` sign.

**Screenshots**

N/A

**Additional context**

N/A

**Acceptance Criteria (for WP Media team use only)**
Clear instructions for developers, to be added before the grooming

### PR DESCRIPTION
**Pull request by remyperona**
## Description

Detect existing fetchpriority attribute when there is spaces around equal sign

Fixes #6325

## Type of change

- Bug fix (non-breaking change which fixes an issue).

# Checklists

## Generic development checklist

- [x] My code follows the style guidelines of this project, with adapted comments and without new warnings.
- [ ] I have added unit and integration tests that prove my fix is effective or that my feature works.
- [ ] The CI passes locally with my changes (including unit tests, integration tests, linter).

## Test summary

- [x] I triggered all changed lines of code at least once without new errors/warnings/notices.

### CODE REVIEW

**Review by CrochetFeve0251 — APPROVED** [review]

File: inc/Engine/Media/AboveTheFold/Frontend/Controller.php [hunk 1]
```diff
@@ -148,7 +148,7 @@ private function set_fetchpriority( $lcp, string $html ): string {
 			'#<img(?:[^>]*?\s+src=["\']' . $url . '["\'][^>]*?|[^>]*?)>#',
 			function ( $matches ) {
 				// Check if the fetchpriority attribute already exists.
-				if ( preg_match( '/fetchpriority=[\'"]([^\'"]+)[\'"]/', $matches[0] ) ) {
+				if ( preg_match( '/fetchpriority\s*=\s*[\'"]([^\'"]+)[\'"]/', $matches[0] ) ) {
```
Comment (wordpressfan) [inline inc/Engine/Media/AboveTheFold/Frontend/Controller.php]: Do we need to handle if the `fetchpriority` is in capital letters like:
https://regex101.com/r/tXEJJN/1

**Comment by vmanthos** [comment]
> Do we need to handle if the fetchpriority is in capital letters

I believe this is always "nice to have" when dealing with text.

**Review by Miraeld — APPROVED** [review]

**Review by wordpressfan — APPROVED** [review]

**Comment by Mai-Saad** [comment]
@Tabrisrp Thanks for the PR most of the cases are working now.
@piotrbak In the following cases we are not adding `fetchpriorty =" high" `in the image markup,  shall we add it? 
1- `Nature` => fetchpriority without any value
2- `` => image name with fetchpriority

**Comment by piotrbak** [comment]
@Mai-Saad That's fine. We need to merge what we have in this approach of 3.16 now to kind of alpha, and we won't be making more fixes here. 

Ping @MathieuLamiot

**Pull request merged**